### ISSUE TITLE
Name is not assigned to Series in binary ops

### ISSUE THREAD

**Issue by gshimansky** [modin-project/modin#1413]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Windows 10

- **Modin version** (`modin.__version__`):

0.7.2+85.gb9b791f.dirty

- **Python version**:

Python 3.7.7

- **Code we can use to reproduce**:

```python
if __name__ == "__main__":
    import numpy as np
    import modin.pandas as pd

    s1 = pd.DataFrame(np.arange(1, 10000, dtype=np.float32), columns=["s1"])
    s2 = np.power(s1['s1'], np.float64(2.0))
    print(s2.name)
```

### Describe the problem

Name from the source Series should be assigned to result, but on Modin result name is `None`.

**Comment by gshimansky** [comment]
The problem with `name` in Series is broader. It is not preserved across all binary operations. The following patch to `test_series.py` makes a lot of tests to fail.
```patch
diff --git a/modin/pandas/test/test_series.py b/modin/pandas/test/test_series.py
index 4369754..a15c901 100644
--- a/modin/pandas/test/test_series.py
+++ b/modin/pandas/test/test_series.py
@@ -169,6 +169,8 @@ def create_test_series(vals):
     elif isinstance(vals, list):
         modin_series = pd.Series(vals)
         pandas_series = pandas.Series(vals)
+    modin_series.name = "series_name"
+    pandas_series.name = modin_series.name
     return modin_series, pandas_series
```

**Comment by gshimansky** [comment]
Commit that attempts to fix this problem is here https://github.com/gshimansky/modin/commit/71db81e11ddbc4ea320bbed498a8a006b3efbe88
There is a problem with it because Pandas apparently is not consistent in how it deals with Series name of binary operations when second operand is an `ndarray`, list or tuple. A bug https://github.com/pandas-dev/pandas/issues/33930 was created on Pandas to address this.

**Comment by prutskov** [comment]
@gshimansky This bug was fixed in #2256. Your reproducer gives result `s1` in current master instead of `None`. This issue can be closed.
